spyder/plugins/editor/panels/scrollflag.py

```diff
@@ -108,58 +147,104 @@ def paintEvent(self, event):
         painter = QPainter(self)
         painter.fillRect(event.rect(), self.editor.sideareas_color)
 
-        # Paint warnings and todos
-        block = self.editor.document().firstBlock()
-        for line_number in range(self.editor.document().blockCount()+1):
-            data = block.userData()
-            if data:
-                if data.code_analysis:
-                    # Paint the warnings
-                    for source, code, severity, message in data.code_analysis:
-                        error = severity == DiagnosticSeverity.ERROR
-                        if error:
-                            painter.setBrush(self._facecolors['error'])
-                            painter.setPen(self._edgecolors['error'])
-                            break
-                    else:
-                        painter.setBrush(self._facecolors['warning'])
-                        painter.setPen(self._edgecolors['warning'])
-
-                    rect_y = self.calcul_flag_ypos(
-                        line_number, scale_factor, offset)
-                    painter.drawRect(rect_x, rect_y, rect_w, rect_h)
-                if data.todo:
-                    # Paint the todos
-                    rect_y = self.calcul_flag_ypos(
-                        line_number, scale_factor, offset)
-                    painter.setBrush(self._facecolors['todo'])
-                    painter.setPen(self._edgecolors['todo'])
-                    painter.drawRect(rect_x, rect_y, rect_w, rect_h)
-                if data.breakpoint:
-                    # Paint the breakpoints
-                    rect_y = self.calcul_flag_ypos(
-                        line_number, scale_factor, offset)
-                    painter.setBrush(self._facecolors['breakpoint'])
-                    painter.setPen(self._edgecolors['breakpoint'])
-                    painter.drawRect(rect_x, rect_y, rect_w, rect_h)
-            block = block.next()
+        editor = self.editor
+        print_local = not bool(self.slider)
+
+        if not print_local:
```

Comment:
For instance, it seems we do not need to compute stuff if the slider scrollbar are is off?

But the code itslef is not enough we really need to document this with comments.

Reply:
Done